docs/manual/working/scalaGuide/main/sql/slick/PlaySlickMigrationGuide.md

```diff
@@ -0,0 +1,106 @@
+
+## `DBAction` and `DBSessionRequest` were removed
+
+Play Slick used to provide a `DBAction` that was useful for:
+
+* Conveniently pass the Slick `Session` into your Action method.
+* Execute the action's body, and hence any blocking call to the database, in a separate thread pool.
+* Limiting the number of blocking requests queued in the thread pool (useful to limit application's latency)
+
+`DBAction` was indeed handy when using Slick 2.1. However, with the new Slick 3 release, we don't need it anymore. The reason is that Slick 3 comes with a new asynchronous API (a.k.a., Database I/O Actions API) that doesn't need the user to manipulate neither a `Session` nor a `Connection`. This makes `DBSessionRequest` and `DBAction`, together with its close friends `CurrentDBAction` and `PredicatedDBAction`, completely obsolete, which is why they have been removed.
+
+Having said that, migrating your code should be as simple as changing all occurrences of `DBAction` and friends, with the standard Play `Action.async`. Click [[here|PlaySlick#Running-a-database-query-in-a-Controller]] for an example.
+
+## Thread Pool
+
+Play Slick used to provide a separate thread pool for executing controller's actions requiring to access a database. Slick 3 already does exactly this, hence there is no longer a need for Play Slick to create and manage additional thread pools. It follows that the below configuration parameter are effectively obsolete and should be removed from your **applications.conf**:
+
+```conf
+db.$dbName.maxQueriesPerRequest
+slick.db.execution.context
+```
+
+The parameter `db.$dbName.maxQueriesPerRequest` was used to limit the number of tasks queued in the thread pool. In Slick 3 you can reach similar results by tuning the configuration parameters `numThreads` and `queueSize`. Read the Slick ScalaDoc for [Database.forConfig].
+
+While the parameter `slick.db.execution.context` was used to name the thread pools created by Play Slick. In Slick 3, each thread pool is named using the Slick database configuration path, i.e., if in your **application.conf** you have provided a Slick configuration for the database named `default`, then Slick will create a thread pool named `default` for executing the database action on the default database. Note that the name used for the thread pool is not configurable.
+
+[Database.forConfig]: http://slick.typesafe.com/doc/3.0.0-RC3/api/index.html#slick.jdbc.JdbcBackend$DatabaseFactoryDef@forConfig(String,Config,Driver):Database
+
+## `Profile` was removed
+
+The trait `Profile` was removed and you can use instead `HasDatabaseConfigProvider` or `HasDatabaseConfig` with similar results.
+
+The trait to use depend on what approach you select to retrieve a Slick database and driver (i.e., an instance of `DatabaseConfig`). If you decide to use dependency injection, then `HasDatabaseConfigProvider` will serve you well. Otherwise, use `HasDatabaseConfig`.
```

Comment:
This and the next section should be moved in the PlaySlick doc page (and here we should add a link to its content)